XIPs/xip-10-brandinfo-schema.md

```diff
@@ -0,0 +1,99 @@
+---
+xip: 10
+title: Brand info schema in conversation metadata
+description: Schema to enable standardized sharing of brand information as it relates to conversationId
+author: Yash Lunagaria (@yash-luna)
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-01-31
+---
+
+## Abstract
+
+This XIP proposes a schema for sharing brand information as it relates to `conversationId`. A standard way to represent this information in conversation `metadata` will enable different front-ends to surface the context captured by `conversationId` in a consistent, user-friendly manner while still affording clients design and UI flexibility.
+
+## Motivation
+
+The SDK API currently accepts a unique `conversationId` per address pair and allows `metadata` to be set as well. Today, different apps use these parameters in different ways and therefore other apps may not know how to render the `conversationId` appropriately within the UI to reflect the corresponding brand. We propose a schema to standardize the use of conversation `metadata` for the purpose of client-side brand expression stemming from the `conversationId`. The `metadata` field may include other non-standard (or future standard) properties.
+
+Note that the schema is only applicable when setting a non-null `conversationId`. If `conversationId` is null, the SDK does not allow setting `metadata`.
+
+## Specification
+
+Proposed `brandInfo` schema in conversation `metadata`
+
+```json
+{
+    conversationId: 'mydomain.xyz/abc/qrs',
+    metadata: {
+        brandInfo: {
+            displayName: 'My Company',
+            profileImage: 'mydomain.xyz/assets/myimage.png,
+            primaryColor: '#ffffff',
+        }
+    }
+}
+```
+
+Example `brandInfo` implementation for a chat app named Galaxy
+
+```json
+{
+    conversationId: 'galaxy.chat/dm/uniqueIdentifier',
+    metadata: {
+        brandInfo: {
+            displayName: 'Galaxy',
+            profileImage: 'galaxychat.com/assets/logo.png,
+            primaryColor: '#6865B8',
+        }
+    }
+}
+```
+
+The `profileImage` file is encouraged to meet the following criteria:
+
+- Aspect ratio: 1:1
+- Minimum resolution: 100x100
+- Maximum resolution: 800x800
```

Comment:
If these are externally hosted, why such a low maximum?

Reply:
Based on Darick's suggestions. Thoughts on increasing min resolution @darickdang?

Reply:
@neekolas Since the image would be used in avatar-like settings (or to complement an avatar), I looked at popular services/libraries such as Gravatar (minimum size 1px) and Libravatar (minimum size is 80x80) standards in addition to Apple's Human Interface Guidelines of having a minimum toucharea of 44x44 (and then I doubled that to add @2x support as the minimum) to create the number you see here.

Open to making it bigger though. Not that strong of an opinion from my end here.

Reply:
I just wonder if some apps are going to want a clickable/zoomed view of app icons. Or an extra large icon. Maybe that's not a real concern here.

Reply:
I wonder if it might help to include a mockup of a UI showing how the `displayName`, `profileImage`, and `primaryColor` might be used? While the `profileImage` might be more obvious, people might be curious about where the `displayName` and `primaryColor` might be used? An image might also really drive home the value! 

And is the idea now that we don't need to use only the `conversationId` as conversation label text?

Reply:
Adding a sample UI is such a good idea @jhaaaa! 

Yes, with a conversationId like 'lens.dev/dm/...' there isn't a shared understanding of how to render it? Should the UI display 'Lens', 'lens.dev', or the complete conversationId? The displayName makes it explicit

Reply:
Great call @jhaaaa. Will look at making an example.